File: test-automation-containers/src/main/java/com/nortal/test/services/testcontainers/ContainerUtils.java
package com.nortal.test.services.testcontainers;

import java.util.List;

import org.apache.commons.lang3.reflect.FieldUtils;
import org.testcontainers.containers.GenericContainer;

public class ContainerUtils {


	public static void overrideNetworkAliases(final GenericContainer<?> genericContainer, final List<String> aliases) {
		try {
			FieldUtils.writeField(genericContainer, "networkAliases", aliases, true);
		} catch (IllegalAccessException e) {
			e.printStackTrace();
		}
	}

}

File: test-automation-containers/src/main/kotlin/com/nortal/test/testcontainers/ContainerUtils.kt
package com.nortal.test.testcontainers

import org.apache.commons.lang3.reflect.FieldUtils
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.testcontainers.containers.GenericContainer

object ContainerUtils {
    private val log: Logger = LoggerFactory.getLogger(javaClass)

    fun overrideNetworkAliases(genericContainer: GenericContainer<*>?, aliases: List<String?>?) {
        try {
            FieldUtils.writeField(genericContainer, "networkAliases", aliases, true)
        } catch (e: IllegalAccessException) {
            log.error("Failed to override network alias", e)
        }
    }
}